docker/bin/zookeeperStart.sh

```diff
@@ -31,101 +32,86 @@ fi
 
 MYID=$((ORD+1))
 
+# Values for first startup
 WRITE_CONFIGURATION=true
 REGISTER_NODE=true
+ONDISK_CONFIG=false
 
 # Check validity of on-disk configuration
 if [ -f $MYID_FILE ]; then
   EXISTING_ID="`cat $DATA_DIR/myid`"
-  if [[ "$EXISTING_ID" == "$MYID" || -f $DYNCONFIG ]]; then
-      WRITE_CONFIGURATION=false
+  if [[ "$EXISTING_ID" == "$MYID" && -f $STATIC_CONFIG ]]; then
+    # If Id is correct and configuration is present under `/data/conf`
+      ONDISK_CONFIG=true
   fi
-
 fi
 
 # Determine if there is a ensemble available to join by checking the service domain
-
 set +e
 nslookup $DOMAIN
 if [[ $? -eq 1 ]]; then
-  set -e
   # If an nslookup of the headless service domain fails, then there is no
   # active ensemble
-  WRITE_CONFIGURATION=true
-  REGISTER_NODE=false
-
+  ACTIVE_ENSEMBLE=false
 else
-  set -e
-  # An ensemble exists, check to see if this node is already a member.
-  set +e
-  ZKURL=$(zkConnectionString)
-  set -e
-  CONFIG=`java -Dlog4j.configuration=file:"$LOG4J_CONF" -jar /root/zu.jar get-all $ZKURL`
-  REGISTERED=`echo "$CONFIG" | grep "server.${MYID}=" | wc -l`
-
-  if [[ $REGISTERED -eq 1 ]]; then
-    REGISTER_NODE=false
+  ACTIVE_ENSEMBLE=true
+fi
 
-  else
-    # When registering the node to the ensemble, always [re]write the config
-    REGISTER_NODE=true
+if [[ "$ONDISK_CONFIG" == true ]]; then
+  # If Configuration is present, we assume, there is no need to write configuration.
+    WRITE_CONFIGURATION=false
+else
     WRITE_CONFIGURATION=true
+fi
 
+if [[ "$ACTIVE_ENSEMBLE" == false ]]; then
+  # This is the first node being added to the cluster
+  REGISTER_NODE=false
+else
+  # ACTIVE_ENSEMBLE == true
```

Comment:
Is this correctly commented out? If so, remove the line.

Reply:
Ok. Done.